.github/workflows/cppinterop-diff.yml

```diff
@@ -0,0 +1,26 @@
+name: Diff CppInterOp against upstream
+on:
+  pull_request:
+    paths:
+      - 'interpreter/CppInterOp/**'
+
+jobs:
+  cppinterop-diff:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out ROOT
+        uses: actions/checkout@v4
+        with:
+          path: root
+      - name: Check out CppInterOp
+        uses: actions/checkout@v4
+        with:
+          repository: compiler-research/CppInterOp
+          ref: be5141cc071841b232e651f173a9759885f31ffb
```

Comment:
As discussed offline, we may want to switch this to tags and potentially a tag file in `interpreter/CppInterOp` (similar to `interpreter/llvm-project/llvm-project.tag`) at some point